app/models/user.rb

```diff
@@ -1,4 +1,16 @@
 class User < ApplicationRecord
   devise :database_authenticatable, :recoverable,
-    :trackable, :validatable, :timeoutable
+    :trackable, :validatable, :timeoutable,
+    :omniauthable, omniauth_providers: [:mojsso]
+
+  def self.from_omniauth(auth)
+    where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
+      user.provider = auth.provider
+      user.uid = auth.uid
+      user.email = auth.info.email
```

Comment:
Yes, there's a unique index for the email:

```sql
Indexes:
    "users_pkey" PRIMARY KEY, btree (id)
    "index_users_on_email" UNIQUE, btree (email)
```

What's the recommended strategy to deal with this cases? Check for user by email first and update only the `provider` and `uid`?